### DISCUSSION TITLE
Proposal: Incorporate Validator Fee into the Validator Trustscore (VTS) Algorithm

### DISCUSSION THREAD

**Discussion by onmax** [nimiq/developer-center#84] [General]
**Original idea:** Micha

## Introduction

The focus on dominance, reliability and availability ensures that validators are evaluated on the basis of their contribution to network decentralisation and performance.

However, I would like to propose the addition of a new parameter: the validator fee. While I understand that this may seem contrary to the original spirit of the VTS - which emphasises the health of the network over individual incentives - I believe that the inclusion of the fee can enhance the usefulness of the score for end users.

## Rationale

- User-centric decision making: Ultimately, stakeholders are interested in maximising their returns while ensuring that their investment makes a positive contribution to the network. If two validators have similar scores, the one with the lower fee will provide a better return for the user.

- Market dynamics: Including the fee in the VTS can encourage healthy competition between validators. Validators may be incentivised to lower their fees to attract more delegations, thereby promoting a more balanced distribution of risk.

- Holistic assessment: While dominance, reliability and availability are critical to the health of the network, the economic aspect cannot be ignored. Fees have a direct impact on a user's reward and should be part of the decision-making process.

## Addressing Potential Conflicts

I acknowledge that adding the fee parameter might seem to conflict with the VTS's goal of focusing solely on network-centric metrics. However, the ultimate purpose of the VTS is to aid users in making informed choices. By ignoring the fee, we might be omitting a critical piece of information that affects user decisions.

## Conclusion

Including the validator fee as a parameter in the VTS algorithm can provide a more comprehensive and user-friendly score. It aligns the VTS more closely with the interests of end users without significantly compromising the original intent of promoting network health and decentralisation.

I invite the community and the development team to review this proposal and provide feedback. Let's work together to improve the VTS for the benefit of all stakeholders.

## Suggestions & Feedback
I'm open to discussing this further and refining the proposal based on community input. Please feel free to share your thoughts, concerns or alternative ideas.

**Comment by sisou** [comment]
Thanks for making me read this very long gpt-generated text...

We currently show the pool's expected reward in the list and in the staking overview, so users _do know_ this number and can compare it. Do you think that is not enough?

**Comment by Eligioo** [comment]
I'm actually against this proposal. In order to provide a reliable staking pool solution (looking from this perspective since the VTS mostly applies to them) you have to make costs for infrastructure, develop and innovate, etc. And in order to cover those costs, you have to apply a fee.

Making the fee part of the VTS will in my opinion result into a race to 0% in order to get a high VTS which stops innovation and result into cheap infrastructure unreliable services.
